### DISCUSSION TITLE
Possible to send request to cluster internal url

### DISCUSSION THREAD

**Discussion by balait4** [Kong/kong#13383] [Help] [answered]
Hi All,

I'm using kong/KIC latest version in okd (openshift) cluster, just want to check is that possible to send request directly to internal cluster url address (ex. kong-proxy-staging.kong.svc.cluster.local). When send to the kong external LB address its working fine. But one requirement where we have 2 application want to talk to each other which want to use the consistent-hashing algorithm in kong and communicate between the 2 pods, so we don't want to go to LB and add latency. So trying to use the kong internal cluster URL . Which is actually not working getting  "message":"no Route matched with those values". So just want to check is this possible or not. Thanks.

**Comment by hanshuebner** [comment, marked as answer]
To Kong Gateway, your internal cluster host name (`kong-proxy-staging.kong.svc.cluster.local`) is just another host name, and if you can send requests to it, Kong can route them.  Kong Gateway does not, however, attach any host names to routes automatically, and if your routes have host names other than your internal cluster host name, they won't match requests with that host name unless you add it explicitly.

Without seeing the details of your configuration, it is not possible to tell whether this is your solution.  Let us know if you need further assistance.

**Reply by balait4** [reply]
Thanks, yes explicitly mentioning the host header worked.
